demos/highlevelapi/src/main/java/org/apache/apex/malhar/stream/sample/WindowedWordCount.java

```diff
@@ -0,0 +1,245 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.stream.sample;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+import org.joda.time.Duration;
+import org.apache.apex.malhar.lib.window.TriggerOption;
+import org.apache.apex.malhar.lib.window.Tuple;
+import org.apache.apex.malhar.lib.window.WindowOption;
+import org.apache.apex.malhar.stream.api.ApexStream;
+import org.apache.apex.malhar.stream.api.WindowedStream;
+import org.apache.apex.malhar.stream.api.function.Function;
+import org.apache.apex.malhar.stream.api.impl.StreamFactory;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.base.Throwables;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.InputOperator;
+import com.datatorrent.api.StreamingApplication;
+import com.datatorrent.api.annotation.ApplicationAnnotation;
+import com.datatorrent.common.util.BaseOperator;
+import com.datatorrent.lib.util.KeyValPair;
+
+import static org.apache.apex.malhar.stream.api.Option.Options.name;
+
+/**
+ * Beam WindowedWordCount Example.
+ */
+@ApplicationAnnotation(name = "WindowedWordCount")
+public class WindowedWordCount implements StreamingApplication
+{
+  static final int WINDOW_SIZE = 1;  // Default window duration in minutes
+  
+  /**
+   * A input operator that reads from and output a file line by line to downstream with a time gap between
+   * every two lines.
+   */
+  public static class TextInput extends BaseOperator implements InputOperator
+  {
+    public final transient DefaultOutputPort<String> output = new DefaultOutputPort<>();
+    
+    private transient BufferedReader reader;
+    
+    @Override
+    public void setup(Context.OperatorContext context)
+    {
+      initReader();
+    }
+    
+    private void initReader()
+    {
+      try {
+        InputStream resourceStream = this.getClass().getResourceAsStream("/wordcount/word.txt");
+        reader = new BufferedReader(new InputStreamReader(resourceStream));
+      } catch (Exception ex) {
+        throw Throwables.propagate(ex);
+      }
+    }
+    
+    @Override
+    public void teardown()
+    {
+      IOUtils.closeQuietly(reader);
+    }
+    
+    @Override
+    public void emitTuples()
+    {
+      try {
+        String line = reader.readLine();
+        if (line == null) {
+          reader.close();
+          initReader();
+        } else {
+          this.output.emit(line);
+        }
+        Thread.sleep(500);
+      } catch (IOException ex) {
+        throw new RuntimeException(ex);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
Not sure whether this falls into the category of checkstyle etc that you asked to exclude, but we should not be printing stack trace to stdout.